Sources/Clappr_iOS/Classes/Plugin/Core/Drawer/DrawerPlugin.swift

```diff
@@ -0,0 +1,82 @@
+open class DrawerPlugin: OverlayPlugin {
+    public enum Position {
+        case undefined
+    }
+
+    open var position: DrawerPlugin.Position {
+        return .undefined
+    }
+
+    open var size: CGSize {
+        return .zero
+    }
+
+    open var placeholder: CGFloat {
+        return .zero
+    }
+
+    private(set) var isClosed: Bool = true {
+        willSet {
+            let event: Event = newValue ? .willHideDrawerPlugin : .willShowDrawerPlugin
+            core?.trigger(event)
+        }
+        didSet {
+            let event: Event = isClosed ? .didHideDrawerPlugin : .didShowDrawerPlugin
+            core?.trigger(event)
+        }
+    }
+
+    public required init(context: UIObject) {
+        super.init(context: context)
+        view.alpha = .zero
+    }
+
+    open override func bindEvents() {
+        bindMediaControlEvents()
+        bindDrawerEvents()
+    }
+
+    private func bindMediaControlEvents() {
+        guard let core = core else { return }
+
+        listenTo(core, event: .willShowMediaControl) { [weak self] _ in
+            UIView.animate(withDuration: ClapprAnimationDuration.mediaControlShow) {
+                self?.view.alpha = 1
+            }
+        }
+
+        listenTo(core, event: .willHideMediaControl) { [weak self] _ in
+            guard self?.isClosed == true else { return }
+            UIView.animate(withDuration: ClapprAnimationDuration.mediaControlHide) {
+                self?.view.alpha = 0
+            }
+        }
+    }
+
+    private func bindDrawerEvents() {
+        guard let core = core else { return }
+
+        listenTo(core, event: .showDrawerPlugin) { [weak self] _ in
+            self?.toggleIsClosed(to: false)
```

Comment:
why not call `self?.showDrawer()` or something like this?

Reply:
There's a reason:

## The current solution:
- `.showDrawerPlugin` event is going to toggle the isClosed flag to `false` _(if it's not false yet)_
- `.hideDrawerPlugin` event is going to toggle the isClosed flag to `true` _(if it's not true yet)_

The function `toggleIsClosed` is handling both cases.

## The proposed change
If we go with the suggested approach, we gonna end up with 2 functions:
- `showDrawer`
- `hideDrawer` 

Both functions are going to duplicate the check if the Bool is already `true` or `false`. 
We've tried this approach before and `CodeClimate` was complaining about it. 


## Conclusion
The Idea was to keep a self explanatory name that reads like an instruction.
if the wording here `toggle "is closed" to false` is not explicit enough, we can come up with a better name for it.